api/v1/submissions/tasks/resources/EditorialTaskParticipantResource.php

```diff
@@ -78,23 +78,33 @@ public function toArray(Request $request, array $data = []): array
             }
         }
 
+        $reviewMethod = null;
         if ($reviewAssignments->isNotEmpty()) {
-            $reviewAssignmentsForUser = $reviewAssignments->where('userId', $this->userId);
-            if ($reviewAssignmentsForUser->isNotEmpty()) {
+            $reviewAssignmentsForUser = $reviewAssignments->first(fn (ReviewAssignment $reviewAssignment) => $reviewAssignment->getReviewerId() == $this->userId);
+            if ($reviewAssignmentsForUser) {
                 $roles[Role::ROLE_ID_REVIEWER] = $reviewerRoleName;
+                $reviewMethods = $reviewAssignments->map(fn (ReviewAssignment $reviewAssignment) => $reviewAssignment->getReviewMethod())->unique();
+                $reviewMethod = $reviewMethods->first(fn () => ReviewAssignment::SUBMISSION_REVIEW_METHOD_DOUBLEANONYMOUS) ??
+                    $reviewMethods->first(fn () => ReviewAssignment::SUBMISSION_REVIEW_METHOD_ANONYMOUS) ??
+                    $reviewMethods->firstWhere(fn () => ReviewAssignment::SUBMISSION_REVIEW_METHOD_OPEN);
             }
         }
 
         $groupedRoles = [];
         foreach ($roles as $roleId => $roleName) {
-            $groupedRoles[] = [
+            $role = [
                 'id' => $roleId,
                 'name' => $roleName,
             ];
+
+            if ($roleId == Role::ROLE_ID_REVIEWER) {
+                $role['reviewMethod'] = $reviewMethod;
+            }
+            $groupedRoles[] = $role;
         }
 
         return [
-            'id' => $this->id,
+            'id' => $this?->id,
```

Comment:
In which cases would `$this` be potentially null?

Reply:
I use the same resource for expose data that isn't yet recorded in the database. Task Participant object is created but not yet saved, so, doesn't have an ID.